jepsen/scalardb/src/scalardb/transfer_append.clj

```diff
@@ -0,0 +1,328 @@
+           (com.scalar.database.transaction.consensuscommit ConsensusCommit)
+           (com.scalar.database.exception.storage NoMutationException)))
+
+(def ^:const KEYSPACE "jepsen")
+(def ^:const TABLE "transfer")
+(def ^:const ACCOUNT_ID "account_id")
+(def ^:const BALANCE "balance")
+(def ^:const AGE "age")
+
+(def ^:const INITIAL_BALANCE 10000)
+(def ^:const NUM_ACCOUNTS 10)
+(def ^:const total-balance (* NUM_ACCOUNTS INITIAL_BALANCE))
+
+(def ^:const NUM_FAILURES_FOR_RECONNECTION 1000)
+
+(defn- create-transfer-table!
+  [session test]
+  (cql/create-keyspace session KEYSPACE
+                       (if-not-exists)
+                       (with {:replication {"class" "SimpleStrategy"
+                                            "replication_factor" (:rf test)}}))
+  (cql/use-keyspace session KEYSPACE)
+  (cql/create-table session TABLE
+                    (if-not-exists)
+                    (column-definitions {:account_id             :int
+                                         :age                    :int
+                                         :balance                :int
+                                         :tx_id                  :text
+                                         :tx_prepared_at         :bigint
+                                         :tx_committed_at        :bigint
+                                         :tx_state               :int
+                                         :tx_version             :int
+                                         :before_balance         :int
+                                         :before_tx_committed_at :bigint
+                                         :before_tx_id           :text
+                                         :before_tx_prepared_at  :bigint
+                                         :before_tx_state        :int
+                                         :before_tx_version      :int
+                                         :primary-key            [:account_id :age]}))
+  (cql/alter-table session TABLE
+                   (with {:compaction-options (c/compaction-strategy)})))
+
+(defn prepare-scan
+  [id]
+  (-> (Key. [(IntValue. ACCOUNT_ID id)])
+      (Scan.)
+      (.forNamespace KEYSPACE)
+      (.forTable TABLE)
+      (.withConsistency Consistency/LINEARIZABLE)
+      (.withOrdering (Scan$Ordering. "age" Scan$Ordering$Order/DESC))
+      (.withLimit 1)))
+
+(defn scan-and-return-result
+  [tx scan]
+  (.get (.scan tx scan) 0))
+
+(defn prepare-get-with-age
+  [id age]
+  (-> ((Key. [(IntValue. ACCOUNT_ID id)]) (Key. [(IntValue. AGE age)]))
+      (Get.)
+      (.forNamespace KEYSPACE)
+      (.forTable TABLE)
+      (.withConsistency Consistency/LINEARIZABLE)))
+
+(defn prepare-put
+  [id age balance]
+  (-> (Put. (Key. [(IntValue. ACCOUNT_ID id)]) (Key. [(IntValue. AGE age)]))
+      (.forNamespace KEYSPACE)
+      (.forTable TABLE)
+      (.withValue (IntValue. BALANCE balance))
+      (.withConsistency Consistency/LINEARIZABLE)))
```

Comment:
I think it's not needed if you use transaction.

Reply:
Thank you. Removed.